From 0aa17bb79dbcdfc249e4841f5b5398e27cfdfd41 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Fri, 15 Nov 2019 11:47:47 +0100
Subject: [PATCH] =?UTF-8?q?pdf-backend:=20fix=20global=20params=20init=20f?=
 =?UTF-8?q?or=20Poppler=20=E2=89=A5=200.83.?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

    pdf-backend.cc: In constructor ‘pdf::Environment::Environment()’:
    pdf-backend.cc:106:35: error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<GlobalParams>’ and ‘GlobalParams*’)
---
 pdf-backend.cc | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/pdf-backend.cc b/pdf-backend.cc
index a94383e..085cd9e 100644
--- pdf-backend.cc
+++ pdf-backend.cc
@@ -101,9 +101,23 @@ static void poppler_error_handler(void *data, ErrorCategory category, pdf::Offse
 }
 #endif
 
+template <typename T> T new_global_params();
+
+// POPPLER_VERSION >= 8300
+template <> std::unique_ptr<GlobalParams> new_global_params<std::unique_ptr<GlobalParams>>()
+{
+  return std::unique_ptr<GlobalParams>(new GlobalParams);
+}
+
+// POPPLER_VERSION < 8300
+template <> GlobalParams* new_global_params<GlobalParams*>()
+{
+  return new GlobalParams;
+}
+
 pdf::Environment::Environment()
 {
-  globalParams = new GlobalParams();
+  globalParams = new_global_params<decltype(globalParams)>();
   setErrorCallback(poppler_error_handler, nullptr);
 }
 
